### ISSUE TITLE
Assertion subject is not validated

### ISSUE THREAD

**Issue by adamaltman** [Redocly/redocly-cli#701]
**Describe the bug**

The assertions subject must be a type (that we defined or a plugin defined). If a subject uses a value that is not defined as a type anywhere, it is likely a mistake. This should result in a warning or an error because the assertion is doing nothing.

How I found it (I made a mistake):

```yaml
    assert/properties-include-examples:
      subject: SchemaProperty
      property: example
      defined: true
      nonEmpty: true
```

"Cool. No issues." Hang on...

```yaml
    assert/properties-include-examples:
      subject: SchemaProperties
      property: example
      defined: true
      nonEmpty: true
```

"Yikes."

The bug here is that there is no warning or error that the SchemaProperty type is not defined.

**To Reproduce**
1. Copy the above into a config.
2. Use any OpenAPI
3. See **no** error

**Expected behavior**

> Subject SchemaProperty does not exist. Did you mean SchemaProperties?

**Logs**
None.

**OpenAPI definition**
3.0

**Redocly Version(s)**
1.0.0-beta.98

**`Node.js` Version(s)**
16

**Additional context**

It seems too easy to make this type of mistake.

**Comment by tatomyr** [comment]
Not only subject. Once I made a typo in a property name...

**Comment by roman-sainchuk** [comment]
This issue depends on [this one](https://github.com/Redocly/redocly-cli/issues/617). So after we make `lint` command to lint  `redocly.yaml` config first (it will lint config by default with possibility to skip it by passing `--skip-config-lint` option), we will be able to add checks for assertion subject and issues will be automatically reported by `lint` command.

**Comment by SmoliyY** [comment]
Closed via #724
